types/delaunator/index.d.ts

```diff
@@ -1,17 +1,19 @@
-// Type definitions for delaunator 2.0
+// Type definitions for delaunator 3.0
 // Project: https://github.com/mapbox/delaunator#readme
 // Definitions by: Denis Carriere <https://github.com/DenisCarriere>
 //                 Bradley Odell <https://github.com/BTOdell>
+//                 Tobias Kraus <https://github.com/tobiaskraus>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare class Delaunator<P> {
+export class Delaunator<P> {
```

Comment:
Should be `export default`

https://github.com/mapbox/delaunator/blob/master/index.js#L5

Reply:
Yes, you are right. And when I test it, in an example, my version doesn't work, but export default works.

But when I change to export default and run `npm run lint delaunator` in DefinitelyTyped, I get this error:

```
Error: C:/.../DefinitelyTyped/types/delaunator/index.d.ts:8:1
ERROR: 8:1  npm-naming  The types for delaunator specify 'export default' but the source does not mention 'default' anywhere.
```

whereas with export class, the lint didn't complain.

Do you know why? What should I do? Ignore the lint?
In the meantime I try to dig deeper by myself...

Reply:
I see that `npm run lint delaunator` looks into `delaunator.min.js` of npms dalaunatator package. (Don't know why, maybe because of package.json "main" attribute) But this file is meant to be required with Node / Browserify. The ES module version lays in `delaunator/index.js`.

Should I simply try to push it, even though the lint fails, or is there anything I can do to fix this lint error message?

Reply:
I tried it anyway (your mentioned changes). But apparently the test breaks, and therefore cannot be merged. The error is the same as I saw locally with `npm run lint delaunator`:
https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/528247076?utm_source=github_status&utm_medium=notification

I need help here.

Reply:
Would need help from the DefinitelyTyped team on this one

Reply:
https://unpkg.com/delaunator@3.0.2/delaunator.js is listed as the package's primary entrypoint, which does not mention a `default`.

Reply:
(It does `return Delaunator;` in a UMD pattern, so an `export = Delaunator` would be appropriate)

Reply:
That's a real tricky one, so that means the Typescript definition would be different for both NodeJS and web?

`module` resolves

```ts
export default class Delaunator {
```

https://unpkg.com/delaunator@3.0.2/index.js

Reply:
Correct, the UMD pattern `export = Delaunator` is correct here.

In NodeJS, if you import via CommonJS, here's what is returned.

```js
> require('delaunator')
{ [Function: Delaunator] from: [Function: from] }
```

Reply:
> That's a real tricky one, so that means the Typescript definition would be different for both NodeJS and web?

@DenisCarriere yes that's how I understand it now as well.

## Another test of mine for Browser (Parcel Bundler):

I tried to change how I understood:

*index.d.ts*
```ts
declare class Delaunator<P> {
  ...
}
export = Delaunator;
```

*delaunator-test.ts*

invalid (tsc complains):
```ts
import Delaunator from 'delaunator';
// error: Module '".../DefinitelyTyped/types/delaunator/index"' has no default export

import { Delaunator } from 'delaunator';
// error: Module '".../DefinitelyTyped/types/delaunator"' has no exported member 'Delaunator'
// error: This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.
```

from https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require I see the right way of importing a module which is exported with `export = ` syntax:
```ts
import Delaunator = require('delaunator');
```

## Result
* `npm run lint delaunator` worked
* an example project of mine with parcel-bundler:
    * `tsc` didn't complain
    * but the compiled JavaScript didn't work because `Delaunator.from` was not a function as `Delaunator` was an object: `{ default: class Delaunator }` 

## Two different definition files ?
So yes, parcel-bundler used the ES module version `delaunator/index.js` version, which requires apparantly a different TypeScript Definition file than Node.js / Browserify needs (for the UMD version)

**What to do? I'm lost. It would have been my first Open Source contribution 😞**

Reply:
@tobiaskraus use the `export =` syntax and just use `esModuleInterop` plus the default imports in your project - that should check out both at compile time and runtime.